### ISSUE TITLE
Jellyfin stuck with 100% cpu load after adding / as library by accident

### ISSUE THREAD

**Issue by flyersa** [jellyfin/jellyfin#12662]
### This issue respects the following points:

- [X] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [X] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [X] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [X] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [X] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

Running  10.9.11. I tried adding a library called /trailers, for some reason the input field auto placed / during removal/readd of the path. Then tried to scan the whole system. After trying to restart jellyfin it is now showing the "select server" panel on the webinterface, when i select my server it says the server is not available. Logs do not show anything meaningfull or any errors, but jellyfin process runs with 100% cpu load.

I tried to add a file /.ignore . Somehow it completly broke down. Is there a way with sqlite3 to check the defined libraries and delete the newly added manually in the sqlite files? I didnt find it in the schema.

I also tried to load with a backup of the library.db, didnt work. so i guess its somewhere in jellyfin.db . How can i enable debug logging?

### Reproduction steps

add a library with / as path

### What is the current _bug_ behavior?

yeah well 100% cpu load and no ui working at all

### What is the expected _correct_ behavior?

yeah to show me the UI

### Jellyfin Server version

10.9.11+

### Specify version number

10.9.11

### Specify the build version

nodashboardworks...

### Environment

```markdown
- OS:
- Linux Kernel:
- Virtualization:
- Clients:
- Browser:
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
[2024-09-17 07:29:16.820 +00:00] [INF] Registering publisher for "urn:schemas-upnp-org:device:MediaServer:1" on "192.168.100.4" with uri "http://192.168.100.4:8096/dlna/aa40ad31-0146-46ce-aa53-16f9f61c06ef/description.xml"
[2024-09-17 07:29:16.829 +00:00] [INF] EventMonitorEntryPoint Running
[2024-09-17 07:29:16.840 +00:00] [INF] Sqlite version: 3.41.2
[2024-09-17 07:29:16.841 +00:00] [INF] Sqlite compiler options: ATOMIC_INTRINSICS=1,COMPILER=gcc-9.4.0,DEFAULT_AUTOVACUUM,DEFAULT_CACHE_SIZE=-2000,DEFAULT_FILE_FORMAT=4,DEFAULT_FOREIGN_KEYS,DEFAULT_JOURNAL_SIZE_LIMIT=-1,DEFAULT_MMAP_SIZE=0,DEFAULT_PAGE_SIZE=4096,DEFAULT_PCACHE_INITSZ=20,DEFAULT_RECURSIVE_TRIGGERS,DEFAULT_SECTOR_SIZE=4096,DEFAULT_SYNCHRONOUS=2,DEFAULT_WAL_AUTOCHECKPOINT=1000,DEFAULT_WAL_SYNCHRONOUS=2,DEFAULT_WORKER_THREADS=0,ENABLE_COLUMN_METADATA,ENABLE_FTS3,ENABLE_FTS3_PARENTHESIS,ENABLE_FTS4,ENABLE_FTS5,ENABLE_MATH_FUNCTIONS,ENABLE_RTREE,ENABLE_SNAPSHOT,MALLOC_SOFT_LIMIT=1024,MAX_ATTACHED=10,MAX_COLUMN=2000,MAX_COMPOUND_SELECT=500,MAX_DEFAULT_PAGE_SIZE=8192,MAX_EXPR_DEPTH=1000,MAX_FUNCTION_ARG=127,MAX_LENGTH=1000000000,MAX_LIKE_PATTERN_LENGTH=50000,MAX_MMAP_SIZE=0x7fff0000,MAX_PAGE_COUNT=1073741823,MAX_PAGE_SIZE=65536,MAX_SQL_LENGTH=1000000000,MAX_TRIGGER_DEPTH=1000,MAX_VARIABLE_NUMBER=32766,MAX_VDBE_OP=250000000,MAX_WORKER_THREADS=8,MUTEX_PTHREADS,SYSTEM_MALLOC,TEMP_STORE=1,THREADSAFE=1
[2024-09-17 07:29:16.854 +00:00] [INF] Default journal_mode for "/var/lib/jellyfin/data/playback_reporting.db" is "delete"
[2024-09-17 07:29:16.855 +00:00] [INF] Initialize PlaybackActivity Repository
[2024-09-17 07:29:16.855 +00:00] [INF] PlaybackActivity table schema OK
[2024-09-17 07:29:16.855 +00:00] [INF] Expected : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2024-09-17 07:29:16.855 +00:00] [INF] Received : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2024-09-17 07:29:17.689 +00:00] [WRN] The WebRootPath was not found: "/var/lib/jellyfin/wwwroot". Static files may be unavailable.
```

**Comment by flyersa** [comment]
btw. "http://$hostname:8096/system/info/public" returns 503 saying "server is starting"

**Comment by flyersa** [comment]
It would be nice if someone can help with the question on how to remove a added library with sqlite already as this seems to be the issue to me. I managed to enable DEBUG logs, but even here is nothing meaningful in here. How to drop a library in sqlite3 ?

**Comment by gnattu** [comment]
At this point I suggest you to start fresh because your database will look like a mess and probably indexed tons of stuff you don't want to index and the time spent on fixing it might be longer than just start fresh.

Everyone can make mistakes and that is why we keep recommending our users to make backup regularly. What you did is something like removing a wrong path in the filesystem and now the OS starts working abnormally.

If you still want to have a try you can open the library.db, got to TypedBaseItems table and find the row with type `MediaBrowser.Controller.Entities.Folder`, and look for its path. If there is something just the root and anything in there you think you don't want you drop them all. 

> like as it is stuck in a loop somewhere

Because indexing the whole root is that slow.

**Comment by flyersa** [comment]
Thats not really helpfull and is indeed a bug. Yes for having backups, but it wasnt me adding / as folder, it was the jellyfin ui when you click on remove. In addition i already tried the library.db backup files, that has no effect at all. 

So thats clearly a bug to me. Deleting the last row in that table has no effect. Also there is no indexing done at all, there is no io load on that process, the behavior is like if you have a while(true) running without any sleep so its maxing the cpu.

Hooray for reindexing 20 TB of data then when i start over...

**Comment by flyersa** [comment]
I deleted another row now  that included basically nothing at all except blanks in the TypedBaseItems and the server started again. Thank you for the hint about that table.

**Comment by gnattu** [comment]
> but it wasnt me adding / as folder, it was the jellyfin ui when you click on remove.

I cannot reproduce such behavior. It will leave a library with 0 folders if you click on the trash bin icon.

You HAVE to:

- Click add
- Click the / folder
- Click OK

To add root as library folder

**Comment by Tukajo** [comment]
I had the same issue. I had jellyfin running via proxmox, and I could not for the life of me figure out what happened to my server after a simple reboot of the container.
**TLDR**
Turns out something (maybe I fat fingered it), I added `/` as a path to my movies folder. And removing that from the library was "not enough".

More information:

Accidentally adding the library path to the system root caused Jellyfin (as mentioned above) to begin monitoring various system paths on my container. And throwing a ton of `UnauthorizedAccessException`s.

I went down the rabbit hole of spinning up a new `Jellyfin` container entirely separately, and verified all of my permissions matched the "brand new instance" (nothing was awry here).

The smoking gun was the `journalctl` and `/var/log/jellyfin/*.log` files showing that the system had begun "monitoring" `/` (the system root).

After I had discovered the erroneous path and removed it, I still could not get Jellyfin to stop monitoring that directory. I eventually found the comment above from @gnattu describing the `TypedBaseItems` table of the library.db. 

Reading that table (`TypedBaseItems`) revealed an ENORMOUSLY large amount of entries. Most of them system paths. I then realized that these were probably all not desirable paths.

I did a quick query to check for paths that matched my specific media folder:
`SELECT type,Path FROM TypedBaseItems WHERE type="MediaBrowser.Controller.Entities.Folder" AND  Path LIKE "/media%";`
Luckily, all of my "good" paths are prefixed by the /media folder.

This revealed the "good entries" and I was able to simply drop all the other entries.
```
DELETE FROM TypedBaseItems
WHERE type = 'MediaBrowser.Controller.Entities.Folder'
  AND Path NOT LIKE '/media%';
```

I am not saying this will "fix it" for everyone, but this did it for me. A combination of that SQL drop statement and the library path removal.

I am not sure what more to say about this, it seems that if the user adds `/` as a library path there should perhaps be a warning that appears as a dialog. I do not know how that got added, but I know I did not do so intentionally.
